extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -498,14 +498,28 @@ export class SafeframeHostApi {
   /**
    * @param {number} height
    * @param {number} width
+   * @param {boolean} isCollapsed
+   * @param {string} messageType
    */
-  resizeIframe(height, width) {
-    if (this.iframe_) {
-      setStyles(this.iframe_, {
-        'height': height + 'px',
-        'width': width + 'px',
-      });
-    }
+  resizeSafeframe(height, width, isCollapsed, messageType) {
+    this.isCollapsed_ = isCollapsed;
+    this.baseInstance_.mutateElement(() => {
+      if (this.iframe_) {
+        setStyles(this.iframe_, {
+          'height': height + 'px',
+          'width': width + 'px',
+        });
+      }
+    }, this.iframe_);
+    // We need to force a measure right now, as without remeasuring, we
+    // will have the wrong size information for the amp-ad and the
+    // safeframe when we send the response message into the safeframe.
+    // This would be an issue specifically for when expand is successful
+    // while amp-ad is out of the viewport, for example.
+    this.baseInstance_.measureElement(() => {
```

Comment:
Move within mutateElement to prevent race condition where iframe is not yet resized yet success message has been sent